pkg/lightning/backend/tidb/tidb.go

```diff
@@ -59,7 +60,7 @@ type tidbRows []tidbRow
 // MarshalLogArray implements the zapcore.ArrayMarshaler interface
 func (rows tidbRows) MarshalLogArray(encoder zapcore.ArrayEncoder) error {
 	for _, r := range rows {
-		encoder.AppendString(string(r))
+		encoder.AppendString(redact.String(string(r)))
```

Comment:
there's a slight difference between this and the original `ZapRedactArray`. 

when redaction is on, the marshal output in `ZapRedactArray` was always `?`, while here we get `?,?,?,?,?`, and thus "leaking" the number of columns. not sure if it is considered sensitive though.

Reply:
Table structure and number of columns seem not sensitive. PTAL @overvenus